### PR TITLE
Fix Build with Dockerfile

### PR DESCRIPTION
**Pull request by alibabakhanlu12**
i wanted to use composer in nexus and i find this repo. i built this app using docker but i run into the problem and build failed . i searched the error i had  and i find that with adding and installing git in dockerfile my problem resolved

This pull request makes the following changes:
* adding git in Dockerfile

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ARG NEXUS_VERSION=latest
 
 FROM maven:3-jdk-8-alpine AS build
```
Comment (stklcode) [inline Dockerfile:5]:
```suggestion
FROM maven:3-eclipse-temurin-17 AS build
```

I'd suggest to drop the `apk add` in favor of switching from alpine to focal Maven image which already contains _git_ (little bigger, but the complete image can be cached)

We should also update it to JDK 17 to be consistent with the build pipeline, although 8 still works for now.

**Review by stklcode — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (CybotTM) [reply Dockerfile:5]: I think you are right, but this PR is just a bugfix, and should not introduce new "features".

**Comment by CybotTM** [comment]
Thanks for this plugin at all and this PR.

Would it be possible to consider renaming this PR to something like 'Fix Build with Docker' to make its purpose even clearer? It helps other to find it while it is not being merged.

Adding the whole error message to make it easier to find when searching for it:

tl;dr: git ... No such file or directory

```
[INFO] Executing: /bin/sh -c cd '/nexus-repository-composer' && 'git' 'rev-parse' '--verify' 'HEAD'
[INFO] Working directory: /nexus-repository-composer
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for org.sonatype.nexus.plugins:composer-parent-parent 0.1.6-SNAPSHOT:
[INFO]
[INFO] org.sonatype.nexus.plugins:composer-parent-parent .. FAILURE [ 17.349 s]
[INFO] nexus-repository-composer .......................... SKIPPED
[INFO] nexus-repository-composer-it ....................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:06 min
[INFO] Finished at: 2024-11-05T07:08:08Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:buildnumber-maven-plugin:1.4:create (build-identity) on project composer-parent: Cannot get the revision information from the scm repository :
[ERROR] Exception while executing SCM command.: Error while executing command. Error while executing process. Cannot run program "git" (in directory "/nexus-repository-composer"): error=2, No such file or directory
```

Additionally, we would be grateful if this PR could be prioritized for merging, as the current Docker build process as documented in the README is unfortunately broken and requires these changes to function correctly again.

Thank you very much for your attention to this!

File: Dockerfile (hunk 1 above)
Comment (stklcode) [reply Dockerfile:5]: I wouldn’t consider streamlining the outdated (2 years) build JDK with the main pipeline a "feature".

https://github.com/sonatype-nexus-community/nexus-repository-composer/blob/80fbc8748d923dc406680360ac0b90c7cac20b9c/.circleci/config.yml#L18

Comment (CybotTM) [reply Dockerfile:5]: I’m not here to debate whether this should be called a feature, update, or enhancement. I just wanted to point out that it’s an unrelated change to the original PR and the specific bug it addresses.

As I’m new to this project, I mean no offense and don’t intend to tell anyone how to manage contributions. I just wanted to bring it to your attention. Thank you! :-)

Comment (stklcode) [reply Dockerfile:5]: It's a different solution for the problem.

The problem is: `maven:3-jdk-8-alpine` is not suited (any more) to build the project without modification.

While yes, installing `git` is a possible solution, it's still a bad one. `maven:3-jdk-8-alpine` is 5(!) years old: Alpine 3.9.4, OpenJDK 1.8.0u212, Maven 3.6.1

I don't really care if we switch to `3-eclipse-temurin-8-alpine` and install git, switch to `3-eclipse-temurin-8` or directly to `eclipse-temurin-17`, but any of these would be better than installing ancient git 2.20 in an outdated build environment.

Comment (CybotTM) [reply Dockerfile:5]: You are right.

**Comment by alibabakhanlu12** [comment]
> Thanks for this plugin at all and this PR.
> 
> Would it be possible to consider renaming this PR to something like 'Fix Build with Docker' to make its purpose even clearer? It helps other to find it while it is not being merged.
> 
> Adding the whole error message to make it easier to find when searching for it:
> 
> tl;dr: git ... No such file or directory
> 
> ```
> [INFO] Executing: /bin/sh -c cd '/nexus-repository-composer' && 'git' 'rev-parse' '--verify' 'HEAD'
> [INFO] Working directory: /nexus-repository-composer
> [INFO] ------------------------------------------------------------------------
> [INFO] Reactor Summary for org.sonatype.nexus.plugins:composer-parent-parent 0.1.6-SNAPSHOT:
> [INFO]
> [INFO] org.sonatype.nexus.plugins:composer-parent-parent .. FAILURE [ 17.349 s]
> [INFO] nexus-repository-composer .......................... SKIPPED
> [INFO] nexus-repository-composer-it ....................... SKIPPED
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time:  01:06 min
> [INFO] Finished at: 2024-11-05T07:08:08Z
> [INFO] ------------------------------------------------------------------------
> [ERROR] Failed to execute goal org.codehaus.mojo:buildnumber-maven-plugin:1.4:create (build-identity) on project composer-parent: Cannot get the revision information from the scm repository :
> [ERROR] Exception while executing SCM command.: Error while executing command. Error while executing process. Cannot run program "git" (in directory "/nexus-repository-composer"): error=2, No such file or directory
> ```
> 
> Additionally, we would be grateful if this PR could be prioritized for merging, as the current Docker build process as documented in the README is unfortunately broken and requires these changes to function correctly again.
> 
> Thank you very much for your attention to this!

**Comment by bhamail** [comment]
Fixed in PR #172
